G/GAP_lib/build_tarballs.jl

```diff
@@ -0,0 +1,65 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, BinaryBuilderBase, Pkg
+
+name = "GAP_lib"
+version = v"4.11.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/gap-system/gap.git", "069a6497424113dc5e5ffe133be9da5bfe6acb24"),
+#    ArchiveSource("https://github.com/gap-system/gap/releases/download/v$(version)/gap-$(version)-core.tar.bz2",
+#                  "6637f66409bc91af21eaa38368153270b71b13b55b75cc1550ed867c629901d1"),
+    ArchiveSource("https://github.com/gap-system/gap/releases/download/v$(version)/packages-required-v$(version).tar.gz",
+                  "29ab6e2752f39d22e3f0a19e5bcbfec1710993b4cfd52337a0ca10fda6a76537";
+                  unpack_target="pkg"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/gap*
+
+mv ../pkg .
+find pkg -name '._*' -exec rm \{\} \; # unwanted files
+
+# run autogen.sh if compiling from it source and/or if configure was patched
+./autogen.sh
+
+# compile a native version of GAP so we can use it to generate the manual
+# (the manual is only in FULL gap release tarballs, not in the -core tarball
+# nor in git snapshots)
+mkdir native-build
+cd native-build
+#apk add gmp-dev zlib-dev
+../configure --build=${MACHTYPE} --host=${MACHTYPE} CC=${CC_BUILD} CXX=${CXX_BUILD} --with-zlib=${prefix} --with-gmp=${prefix}
+make -j${nproc}
+make html   # build the manual (only HTML and txt; for PDF we'd need LaTeX)
+cd ..
+
+# remove the native build, it has done its job
+rm -rf native-build
+rm -rf autom4te.cache dev extern hpcgap/extern pkg
+
+install_license LICENSE
+
+# package up everything into the worktree
+mkdir -p ${prefix}/gap/
+mv * ${prefix}/gap/
```

Comment:
Not super important, but any particular reason for installing this in `${prefix}/gap` rather than, for example `${prefix}/share/gap`?

Reply:
I doesn't really matter. I am happy to move it :-)

Reply:
I don't have a strong opinion about that, just found a bit surprising to install something outside of the "standard" directories in the prefix, but it doesn't really matter much

Reply:
Yeah, I moved it but ... 🤷‍♂️ it really is just a glorified data tarball, and I abuse Yggdrasil to perform some preprocessing in an automated fashion, which simplifies life a *ton* for me, and it is super convenient that I can have both the binaries and data files in JLLs (GAP_jll and GAP_lib_jll), that really helps

Reply:
If this isn't needed as build dependency for anything else, you could also have created the artifact locally, for example with [`ArtifactUtils.jl`](https://github.com/simeonschaub/ArtifactUtils.jl).  But ok, this is merged already :sweat_smile:

Reply:
If this was a static thing, I'd do that -- and I *did* in the past. But it requires non-trivial code execution and data massaging, and only I knew the exact steps (and if you only do it every couple month, you forget).

With this setup, an update is trivial and also easy to teach my coworkers how to do it.

So the real alternative would be to set up some other CI solution somewhere... if it is deemed better to do, I can look into transferring this at some point. For now it is super convenient:-)